rest-api/api/base_api.py

```diff
@@ -132,7 +134,7 @@ def _make_query(self):
     return Query(field_filters, order_by, max_results, pagination_token,
                  include_total=include_total)
 
-  def _make_bundle(self, results, id_field, participant_id):
+  def _make_bundle(self, results, id_field, participant_id, csv=False):
```

Comment:
So _make_bundle constructs a JSON object with an array of entries, each one containing a resource JSON dictionary, which gets returned as JSON to the client. My assumption was that we were going to change the whole response to be CSV, not a JSON object containing CSV... which would be what this code would do I think. So I don't think you want to mess with _make_bundle.

Instead, just have a whole new _make_csv(results) method that makes a CSV writer, calls to_client_csv on each object (passing in the writer), and then returns a Response for the whole thing. You'd call that from participant_summary_api.get() when output=csv.

If you want to stream the results to the client as they are written, this may give some indication of how to do that:

https://stackoverflow.com/questions/32608265/streaming-a-generated-csv-with-flask

A baby step I guess that would be a bit easier to implement would be to have the "resource" in each item be a JSON array of items, rather than a JSON dict like today. I guess we could test how much faster *that* makes things run, if we didn't want to return actual CSV. But it seems a bit kooky. :)

Reply:
Note you'll need the writer to first spit out the column headers, which means figuring out what order columns are going to be returned and making sure that matches between the headers and your to_client_csv code.